frontend/components/Signup.js

```diff
@@ -0,0 +1,170 @@
+import React from "react";
+import {
+  StyleSheet,  
+  Text,
+  View,
+  TextInput,
+  TouchableOpacity,
+} from "react-native";
+
+export default class Signup extends React.Component{
+    constructor(props) {
+        super(props);
+        this.state = {
+          fullname: "",
+          email:"",
+          phonenumber:"",
+          password: "",
+          confirmpassword:"",
+        }
+      }
+      validate=()=>
+      {
+        const{fullname,email,phonenumber,password,confirmpassword}=this.state
+        if(fullname=="")
```

Comment:
Instead of first name there should be first name and last name
You haven't used the trim function to remove spaces from string